### PR TITLE
monitor: WLS workaround for ConsoleReader cancel

### PR DESCRIPTION
**Pull request by enrikb**
This patch addresses the fact that TIOCSTI is not available on WSL /
bash on windows. It avoids calling getkey() when no data is available
very much like the workaround already available for Windows.

(Note that pthread_kill() is not available before python 3.3 and serial
read with timeout (VMIN = 0, VTIME = 1) does not work on WSL, either.
Therefore, the workaround was implemented using peek/wait.)

### CODE REVIEW

File: tools/idf_monitor.py [hunk 1]
```diff
@@ -159,7 +179,9 @@ def run(self):
             self.console.cleanup()
 
     def _cancel(self):
-        if os.name == 'posix':
+        if self.wsl:
+            pass
```
Comment (projectgus) [inline tools/idf_monitor.py]: What do you think about `if os.name == 'posix' and not self.is_wsl` ?

Comment (enrikb) [reply tools/idf_monitor.py]: I would prefer to keep the WSL workaround separate from the original implementation, as the original will not introduce extra latency when typing on the console (once stdin from console will be implemented ...).

Comment (enrikb) [reply tools/idf_monitor.py]: Uh, sorry, I misunderstood your suggestion.
I hope I got it now. I have slightly reworked the variable naming, changed the condition like you suggested and already squashed/rebased the commit.

**Comment by enrikb** [comment]
Unfortunately, the method used in my change does not work at all due to stdin etc. buffering. For example, escape sequences will starve in the input buffer. Most probably select() has been removed from the original miniterm for the same reason, as mentioned in the comments.

At this time, I will cancel the pull request until I will hopefully find a better solution.

**Pull request closed without merging**